src/benchmarkstt/metrics/core.py

```diff
@@ -159,6 +160,157 @@ def compare(self, ref: Schema, hyp: Schema) -> OpcodeCounts:
         return get_opcode_counts(diffs.get_opcodes())
 
 
+class BEER(Metric):
+    """
+    Bag of Entities Error Rate, BEER, is defined as the error rate per entity with a bag of words approach::
+
+                            abs(ne_hyp - ne_ref)
+        BEER (entity)   =   ----------------------
+                                ne_ref
+
+    - ne_hyp = number of detections of the entity in the hypothesis file
+    - ne_ref = number of detections of the entity in the reference file
+
+    The WA_BEER for a set of N entities is defined as the weighted average of the BEER for the set of
+    entities::
+
+        WA_BEER ([entity_1, ... entity_N) =  w_1*BEER (entity_1)*L_1/L + ... + w_N*BEER (entity_N))*L_N/L
+
+    which is equivalent to::
+
+                                            w_1*abs(ne_hyp_1 - ne_ref_1) + ... + w_N*abs(ne_hyp_N - ne_ref_N)
+        WA_BEER ([entity_1, ... entity_N) = ------------------------------------------------------------------
+                                                                            L
+
+    - L_1 =  number of occurrences of entity 1 in the reference document
+    - L = L_1 + ... + L_N
+
+    the weights being normalised by the tool:
+
+    - w_1 + ... + w_N = 1
+
+    The input file defines the list of entities and the weight per entity, w_n. It is processed as a json file with the
+    following structure::
+
+        { "entity_1":W_1, "entity_2" : W_2, "entity_3" :W_3 .. }
+
+    W_n being the non-normalized weight, the normalization of the weights is performed by the tool as::
+
+                    W_n
+        w_n =   ---------------
+                W_1 + ... +W_N
+
+    The minimum value for weight being 0.
+
+    """
+    def __init__(self, entities_file=''):
+        """
+        """
+        self._weight = []
+        self._entities = []
+
+        try:
+            with open(entities_file) as f:
+                data = json.load(f)
+            self._entities = list(data.keys())
+            weight = list(data.values())
+            self.set_weight(weight)
+        except IOError:
+            print('Input file Error')
```

Comment:
Originally it generated an empty dict in case of invalid file, for sake of readability I put the reading of the file in the __init__(). If I move it to compare() it generates issues for testing. I would suggest generating an error that break the pipe or an empty dict as done in the first version.